src/probnum/filtsmooth/statespace/generate_samples.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+
+def generate_samples(dynmod, measmod, initrv, times, num_steps=5):
```

Comment:
Good question. I tend to use it quite heavily in examples and tests, so I think would be a bit awkward to have it as `_generate_samples`. 
I also changed the name because `generate` was a bit cryptic IMO if encountered in the context of tests and examples.